app/src/app/views/stix/stix-dialog/stix-dialog.component.ts

```diff
@@ -19,15 +19,27 @@ export class StixDialogComponent implements OnInit {
         if (this._config.mode && this._config.mode == "edit") this.startEditing();
     }
     public get config(): StixViewConfig {
+        let object = Array.isArray(this._config.object)? this._config.object[0] : this._config.object;
         return {
             mode: this.editing? "edit" : "view",
-            object: this._config.object,
-            showRelationships: false,
+            object: object,
+            showRelationships: object.attackType == "data-component" ? true : false,
             editable: this._config.editable,
-            sidebarControl: this._config.sidebarControl == "disable"? "disable" : "events"
+            sidebarControl: this._config.sidebarControl == "disable"? "disable" : "events",
+            dialog: this.dialogRef
         }
     }
 
+    public prevObject;
+    public openObject(object: StixObject): void {
+        this.prevObject = this._config.object;
+        this._config.object = object;
+    }
```

Comment:
This is poorly documented and the function name is very vague. I understand what's going on only because I've read the rest of the code in this PR, but I wouldn't expect future maintainers to understand just from seeing this component.

Reply:
I don't know if this is the case or not, but is it theoretically possible to chain `openObject()` calls to walk a (conceptual) link list of dialog contents? I think `prevObject` is too limited for this behavior to work properly but from the API of stix-dialog it is not apparent that this won't work.